src/lib/crypto/kimchi_backend/gadgets/foreign_field.ml

```diff
@@ -1600,30 +1599,21 @@ let bytes_to_standard_element (type f)
     standard_element_as_linear_combination_of_bytes (module Circuit) bytestring
   in
 
-  (* Check modulus_bit_length = # of bits you unpack
-   * This is partly implicit in the circuit given the number of byte outputs of Keccak:
-   * · input_bitlen < fmod_bitlen : OK
-   * · input_bitlen = fmod_bitlen : OK
-   * · input_bitlen > fmod_bitlen : CONSTRAIN
-   * If bit length of unpacked bits is at most fmod_bitlen,
-   * then elem will be at most twice the foreign modulus.
-   * and one single reduction with FFAdd will be enough.
-   *)
+  (* Check that input is in correct limb shape is implicit from linear combination of bytes *)
+  (* Meaning, no need to add a multi range check for elem *)
 
   (* C2: Reduce z modulo foreign_field_modulus
    *
-   *   Constrain z' = z + 0 modulo foreign_field_modulus using foreign field addition gate
+   *   Constrain z = z' mod foreign_field_modulus using foreign field multiplication gate
+   *
+   *   z * 1 = q * f + z' 
    *
-   *   Note: doing this once is sufficient because z cannot be larger than double the size due to bit length constraint
+   *   Note: any size (up to 2^259-1) is supported using this kind of reduction
    *)
-  let zero = Element.Standard.of_limbs (Field.zero, Field.zero, Field.zero) in
-  (* C3: Range check z' < f *)
-  (* Altogether this is a call to Foreign_field.add in final=true mode *)
-  (* with a canonical check for the output *)
-  let output =
-    add (module Circuit) ~final:true external_checks elem zero fmod
-  in
+  let one = Element.Standard.of_limbs (Field.zero, Field.zero, Field.one) in
```

Comment:
If I'm not wrong, the limbs start with the smallest, so 1 would be (one, zero, zero), see e.g. https://github.com/MinaProtocol/mina/blob/eb157617909a5aef4db12c24c80d3315b73f3a4e/src/lib/crypto/kimchi_backend/gadgets/foreign_field.ml#L304-L316

but you can also use `Foreign_field.one`

Reply:
My bad, I was doing some boolean tasks in between, in big endian 😅 , and forgot to check the order

Reply:
I just added a parameter to the tests with an optional "expected" BigInt value to make sure that it is actually converting the value as expected (having had it before would have caught that it was using 2^176 value instead of 1).